src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -471,6 +444,82 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
+func redpandaEnvVars() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "REDPANDA_ENVIRONMENT",
+			Value: "kubernetes",
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "status.podIP",
+				},
+			},
+		},
+	}
+}
+
+func (r *StatefulSetResource) wasmContainer() corev1.Container {
+	if r.pandaCluster.Spec.DeepCopy().ResourcesWASM == nil {
+		r.logger.Info("BUG! No resources found for wasm - this should never happen with defaulting webhook enabled")
+		r.pandaCluster.Spec.ResourcesWASM = redpandav1alpha1.DefaultWASMResources
+	}
+	return corev1.Container{
+		Name:      wasmContainerName,
+		Image:     r.pandaCluster.FullImageName(),
+		Command:   []string{"/usr/local/bin/coproc.sh"},
+		Env:       redpandaEnvVars(),
```

Comment:
I don't understand what environment variable is needed for `coproc`? To be honest I don't know if any of the env vars are needed for Redpanda as all the heavy lifting is done by configurator process.